[BUG] Unable to compile on Linux 6.14 due to "error: field 'platform_profile_handler' has incomplete type"

**Problem Description**
This module is unable to compile on Linux 6.14 Zen NixOS.

**Model and Debug Info**

Installation type: NixOS Flake
Distribution: NixOS 6.14.0-zen1
Model name: 16ach5h
CPU model: 5800h
GPU model: 3060 Mobile
Keyboard backlight: Single Color
Light in lid or logo: Yes
Light at IO-Ports at back: No

**The below output was done on LQX 6.13 as Zen 6.14 doesn't boot.**

Output of `sudo dmidecode -t system`. Please remove Serial Number and UUID for privacy:
```text
# dmidecode 3.6
Getting SMBIOS data from sysfs.
SMBIOS 3.3.0 present.

Handle 0x0001, DMI type 1, 27 bytes
System Information
	Manufacturer: LENOVO
	Product Name: 82JU
	Version: Legion 5 15ACH6H
	Serial Number:
	UUID:
	Wake-up Type: Power Switch
	SKU Number: LENOVO_MT_82JU_BU_idea_FM_Legion 5 15ACH6H
	Family: Legion 5 15ACH6H

Handle 0x0020, DMI type 12, 5 bytes
System Configuration Options
	Option 1: String1 for Type12 Equipment Manufacturer
	Option 2: String2 for Type12 Equipment Manufacturer
	Option 3: String3 for Type12 Equipment Manufacturer
	Option 4: String4 for Type12 Equipment Manufacturer

Handle 0x0030, DMI type 32, 20 bytes
System Boot Information
	Status: No errors detected
```

Output of `sudo dmidecode -t bios`:
```text
# dmidecode 3.6
Getting SMBIOS data from sysfs.
SMBIOS 3.3.0 present.

Handle 0x0000, DMI type 0, 26 bytes
BIOS Information
	Vendor: LENOVO
	Version: GKCN65WW
	Release Date: 01/16/2024
	Address: 0xE0000
	Runtime Size: 128 kB
	ROM Size: 16 MB
	Characteristics:
		PCI is supported
		BIOS is upgradeable
		BIOS shadowing is allowed
		Boot from CD is supported
		Selectable boot is supported
		EDD is supported
		Japanese floppy for NEC 9800 1.2 MB is supported (int 13h)
		Japanese floppy for Toshiba 1.2 MB is supported (int 13h)
		5.25"/360 kB floppy services are supported (int 13h)
		5.25"/1.2 MB floppy services are supported (int 13h)
		3.5"/720 kB floppy services are supported (int 13h)
		3.5"/2.88 MB floppy services are supported (int 13h)
		8042 keyboard services are supported (int 9h)
		CGA/mono video services are supported (int 10h)
		ACPI is supported
		USB legacy is supported
		BIOS boot specification is supported
		Targeted content distribution is supported
		UEFI is supported
	BIOS Revision: 1.65
	Firmware Revision: 1.65

Handle 0x0021, DMI type 13, 22 bytes
BIOS Language Information
	Language Description Format: Long
	Installable Languages: 4
		en|US|iso8859-1
		fr|FR|iso8859-1
		ja|JP|unicode
		zh|TW|unicode
	Currently Installed Language: en|US|iso8859-1
```

Output of `sudo cat /sys/kernel/debug/legion/fancurve`:
```text
EC Chip ID: 8227
EC Chip Version: 2a4
legion_laptop features: fancurve powermode platformprofile platformprofilenotify minifancurve fancurve_pmw_speed fancurve_rpm_speed
legion_laptop ec_readonly: 0
ACPI CFG error: 0
ACPI CFG: 2081289482
temperature access method: 1
CPU temperature error: 0
CPU temperature: 45
CPU temperature EC error: 0
CPU temperature EC: 45
CPU temperature ACPI error: 0
CPU temperature ACPI: 45
CPU temperature WMI error: 0
CPU temperature WMI: 0
CPU temperature WMI2 error: 0
CPU temperature WMI2: 45
CPU temperature WMI3 error: 0
CPU temperature WMI3: 0
GPU temperature error: 0
GPU temperature: 40
GPU temperature EC error: 0
GPU temperature EC: 40
GPU temperature ACPI error: 0
GPU temperature ACPI: 40
GPU temperature WMI error: 0
GPU temperature WMI: 0
GPU temperature WMI2 error: 0
GPU temperature WMI2: 40
GPU temperature WMI3 error: 0
GPU temperature WMI3: 0
fan speed access method: 1
1 fanspeed error: 0
1 fanspeed: 0
1 fanspeed EC error: 0
1 fanspeed EC: 0
1 fanspeed ACPI error: 0
1 fanspeed ACPI: 0
1 fanspeed WMI error: 0
1 fanspeed WMI: 0
1 fanspeed WMI2 error: 0
1 fanspeed WMI2: 0
1 fanspeed WMI3 error: 0
1 fanspeed WMI3: 0
2 fanspeed error: 0
2 fanspeed: 0
2 fanspeed EC error: 0
2 fanspeed EC: 0
2 fanspeed ACPI error: 0
2 fanspeed ACPI: 0
2 fanspeed WMI error: 0
2 fanspeed WMI: 0
2 fanspeed WMI2 error: 0
2 fanspeed WMI2: 0
2 fanspeed WMI3 error: 0
2 fanspeed WMI3: 0
powermode access method: 3
powermode error: 0
powermode: 3
powermode EC error: 0
powermode EC: 17
powermode ACPI error: -5
powermode ACPI: 0
powermode WMI error: 0
powermode WMI: 3
has custom powermode: 1
ACPI rapidcharge error: 0
ACPI rapidcharge: 0
WMI backlight 2 state: 0
WMI backlight 3 state: 0
WMI light IO port: 0
WMI light Y logo/lid: 0
EC minifancurve feature enabled: 1
EC minifancurve on cool: true
EC lockfancontroller error: 0
EC lockfancontroller: false
fanfullspeed error: 0
fanfullspeed: 0
fanfullspeed EC error: 0
fanfullspeed EC: 0
Max speed for fancurve: 10000
Current fan curve in hardware:
Fan curve current point id: 0
Fan curve points size: 10
u(speed_of_unit)|speed1[u]|speed2[u]|speed1[pwm]|speed2[pwm]|acceleration|deceleration|cpu_min_temp|cpu_max_temp|gpu_min_temp|gpu_max_temp|ic_min_temp|ic_max_temp
3	0	0	0	0	2	2	0	59	0	62	0	37
3	16	16	40	40	2	2	54	59	59	62	33	41
3	18	18	45	45	2	2	54	59	59	62	38	46
3	21	21	53	53	2	2	54	62	59	62	41	127
3	23	23	58	58	2	2	58	68	59	62	127	127
3	26	26	66	66	2	2	64	72	59	62	127	127
3	29	29	73	73	2	2	68	78	59	68	127	127
3	35	35	89	89	2	2	72	85	64	76	127	127
3	42	42	107	107	2	2	80	90	70	80	127	127
3	45	45	114	114	2	2	85	127	76	127	127	127
=====================
Current fan curve in hardware (WMI; might be empty)
Fan curve current point id: 0
Fan curve points size: 0
u(speed_of_unit)|speed1[u]|speed2[u]|speed1[pwm]|speed2[pwm]|acceleration|deceleration|cpu_min_temp|cpu_max_temp|gpu_min_temp|gpu_max_temp|ic_min_temp|ic_max_temp
=====================
```

Output of `nix log /nix/store/m9jagd02crxrfq5rpqis2ya4a38jv9hr-lenovo-legion-module-0.0.19-unstable-2025-02-20.drv`

``` text
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/dyqy3cmnghq772cbfa5vfn1gw8paf2qd-source
source root is source/kernel_module
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
no configure script, doing nothing
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
build flags: SHELL=/nix/store/58br4vk3q5akf4g8lx0pqzfhn47k3j8d-bash-5.2p37/bin/bash ARCH=x86_64 CROSS_COMPILE= KBUILD_OUTPUT=/nix/store/dj7793gb47s632rap69mhigq6lfca5ld-linux-zen-6.14-dev/lib/modules/6.14.0-zen1/build SHELL=bash KERNELVERSION=6.14.0-zen1 KSRC=/nix/store/dj7793gb47s632rap69mhigq6lfca5ld-linux-zen-6.14-dev/lib/modules/6.14.0-zen1/build INSTALLDIR=/nix/store/d1317rwsw38qlvvgjz7jxb9hyn56770v-lenovo-legion-module-0.0.19-unstable-2025-02-20/lib/modules/6.14.0-zen1/kernel/drivers/platform/x86 MODDESTDIR=/nix/store/d1317rwsw38qlvvgjz7jxb9hyn56770v-lenovo-legion-module-0.0.19-unstable-2025-02-20/lib/modules/6.14.0-zen1/kernel/drivers/platform/x86 DKMSDIR=/nix/store/d1317rwsw38qlvvgjz7jxb9hyn56770v-lenovo-legion-module-0.0.19-unstable-2025-02-20/lib/modules/6.14.0-zen1/misc
make -C /nix/store/dj7793gb47s632rap69mhigq6lfca5ld-linux-zen-6.14-dev/lib/modules/6.14.0-zen1/build M=/build/source/kernel_module modules
make[1]: Entering directory '/nix/store/dj7793gb47s632rap69mhigq6lfca5ld-linux-zen-6.14-dev/lib/modules/6.14.0-zen1/build'
make[2]: Entering directory '/build/source/kernel_module'
  CC [M]  legion-laptop.o
[01m[Klegion-laptop.c:2411:41:[m[K [01;31m[Kerror: [m[Kfield '[01m[Kplatform_profile_handler[m[K' has incomplete type
 2411 |         struct platform_profile_handler [01;31m[Kplatform_profile_handler[m[K;
      |                                         [01;31m[K^~~~~~~~~~~~~~~~~~~~~~~~[m[K
[01m[Klegion-laptop.c:[m[K In function '[01m[Klegion_platform_profile_notify[m[K':
[01m[Klegion-laptop.c:4888:9:[m[K [01;31m[Kerror: [m[Ktoo few arguments to function '[01m[Kplatform_profile_notify[m[K'
 4888 |         [01;31m[Kplatform_profile_notify[m[K();
      |         [01;31m[K^~~~~~~~~~~~~~~~~~~~~~~[m[K
In file included from [01m[Klegion-laptop.c:74[m[K:
[01m[K/nix/store/dj7793gb47s632rap69mhigq6lfca5ld-linux-zen-6.14-dev/lib/modules/6.14.0-zen1/source/include/linux/platform_profile.h:58:6:[m[K [01;36m[Knote: [m[Kdeclared here
   58 | void [01;36m[Kplatform_profile_notify[m[K(struct device *dev);
      |      [01;36m[K^~~~~~~~~~~~~~~~~~~~~~~[m[K
In file included from [01m[K/nix/store/dj7793gb47s632rap69mhigq6lfca5ld-linux-zen-6.14-dev/lib/modules/6.14.0-zen1/source/include/linux/bits.h:22[m[K,
                 from [01m[K/nix/store/dj7793gb47s632rap69mhigq6lfca5ld-linux-zen-6.14-dev/lib/modules/6.14.0-zen1/source/include/linux/ioport.h:13[m[K,
                 from [01m[K/nix/store/dj7793gb47s632rap69mhigq6lfca5ld-linux-zen-6.14-dev/lib/modules/6.14.0-zen1/source/include/linux/acpi.h:12[m[K,
                 from [01m[Klegion-laptop.c:62[m[K:
[01m[Klegion-laptop.c:[m[K In function '[01m[Klegion_platform_profile_get[m[K':
[01m[K/nix/store/dj7793gb47s632rap69mhigq6lfca5ld-linux-zen-6.14-dev/lib/modules/6.14.0-zen1/source/include/linux/compiler_types.h:483:27:[m[K [01;31m[Kerror: [m[Kexpression in static assertion is not an integer
  483 | #define __same_type(a, b) [01;31m[K__builtin_types_compatible_p[m[K(typeof(a), typeof(b))
      |                           [01;31m[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~[m[K
[01m[K/nix/store/dj7793gb47s632rap69mhigq6lfca5ld-linux-zen-6.14-dev/lib/modules/6.14.0-zen1/source/include/linux/build_bug.h:78:56:[m[K [01;36m[Knote: [m[Kin definition of macro '[01m[K__static_assert[m[K'
   78 | #define __static_assert(expr, msg, ...) _Static_assert([01;36m[Kexpr[m[K, msg)
      |                                                        [01;36m[K^~~~[m[K
[01m[K/nix/store/dj7793gb47s632rap69mhigq6lfca5ld-linux-zen-6.14-dev/lib/modules/6.14.0-zen1/source/include/linux/container_of.h:20:9:[m[K [01;36m[Knote: [m[Kin expansion of macro '[01m[Kstatic_assert[m[K'
   20 |         [01;36m[Kstatic_assert[m[K(__same_type(*(ptr), ((type *)0)->member) ||       \
      |         [01;36m[K^~~~~~~~~~~~~[m[K
[01m[K/nix/store/dj7793gb47s632rap69mhigq6lfca5ld-linux-zen-6.14-dev/lib/modules/6.14.0-zen1/source/include/linux/container_of.h:20:23:[m[K [01;36m[Knote: [m[Kin expansion of macro '[01m[K__same_type[m[K'
   20 |         static_assert([01;36m[K__same_type[m[K(*(ptr), ((type *)0)->member) ||       \
      |                       [01;36m[K^~~~~~~~~~~[m[K
[01m[Klegion-laptop.c:4897:16:[m[K [01;36m[Knote: [m[Kin expansion of macro '[01m[Kcontainer_of[m[K'
 4897 |         priv = [01;36m[Kcontainer_of[m[K(pprof, struct legion_private,
      |                [01;36m[K^~~~~~~~~~~~[m[K
[01m[Klegion-laptop.c:[m[K In function '[01m[Klegion_platform_profile_set[m[K':
[01m[K/nix/store/dj7793gb47s632rap69mhigq6lfca5ld-linux-zen-6.14-dev/lib/modules/6.14.0-zen1/source/include/linux/compiler_types.h:483:27:[m[K [01;31m[Kerror: [m[Kexpression in static assertion is not an integer
  483 | #define __same_type(a, b) [01;31m[K__builtin_types_compatible_p[m[K(typeof(a), typeof(b))
      |                           [01;31m[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~[m[K
[01m[K/nix/store/dj7793gb47s632rap69mhigq6lfca5ld-linux-zen-6.14-dev/lib/modules/6.14.0-zen1/source/include/linux/build_bug.h:78:56:[m[K [01;36m[Knote: [m[Kin definition of macro '[01m[K__static_assert[m[K'
   78 | #define __static_assert(expr, msg, ...) _Static_assert([01;36m[Kexpr[m[K, msg)
      |                                                        [01;36m[K^~~~[m[K
[01m[K/nix/store/dj7793gb47s632rap69mhigq6lfca5ld-linux-zen-6.14-dev/lib/modules/6.14.0-zen1/source/include/linux/container_of.h:20:9:[m[K [01;36m[Knote: [m[Kin expansion of macro '[01m[Kstatic_assert[m[K'
   20 |         [01;36m[Kstatic_assert[m[K(__same_type(*(ptr), ((type *)0)->member) ||       \
      |         [01;36m[K^~~~~~~~~~~~~[m[K
[01m[K/nix/store/dj7793gb47s632rap69mhigq6lfca5ld-linux-zen-6.14-dev/lib/modules/6.14.0-zen1/source/include/linux/container_of.h:20:23:[m[K [01;36m[Knote: [m[Kin expansion of macro '[01m[K__same_type[m[K'
   20 |         static_assert([01;36m[K__same_type[m[K(*(ptr), ((type *)0)->member) ||       \
      |                       [01;36m[K^~~~~~~~~~~[m[K
[01m[Klegion-laptop.c:4926:16:[m[K [01;36m[Knote: [m[Kin expansion of macro '[01m[Kcontainer_of[m[K'
 4926 |         priv = [01;36m[Kcontainer_of[m[K(pprof, struct legion_private,
      |                [01;36m[K^~~~~~~~~~~~[m[K
[01m[Klegion-laptop.c:[m[K In function '[01m[Klegion_platform_profile_init[m[K':
[01m[Klegion-laptop.c:4974:15:[m[K [01;31m[Kerror: [m[Ktoo few arguments to function '[01m[Kplatform_profile_register[m[K'
 4974 |         err = [01;31m[Kplatform_profile_register[m[K(&priv->platform_profile_handler);
      |               [01;31m[K^~~~~~~~~~~~~~~~~~~~~~~~~[m[K
[01m[K/nix/store/dj7793gb47s632rap69mhigq6lfca5ld-linux-zen-6.14-dev/lib/modules/6.14.0-zen1/source/include/linux/platform_profile.h:50:16:[m[K [01;36m[Knote: [m[Kdeclared here
   50 | struct device *[01;36m[Kplatform_profile_register[m[K(struct device *dev, const char *name,
      |                [01;36m[K^~~~~~~~~~~~~~~~~~~~~~~~~[m[K
[01m[Klegion-laptop.c:[m[K In function '[01m[Klegion_platform_profile_exit[m[K':
[01m[Klegion-laptop.c:4988:9:[m[K [01;31m[Kerror: [m[Ktoo few arguments to function '[01m[Kplatform_profile_remove[m[K'
 4988 |         [01;31m[Kplatform_profile_remove[m[K();
      |         [01;31m[K^~~~~~~~~~~~~~~~~~~~~~~[m[K
[01m[K/nix/store/dj7793gb47s632rap69mhigq6lfca5ld-linux-zen-6.14-dev/lib/modules/6.14.0-zen1/source/include/linux/platform_profile.h:53:5:[m[K [01;36m[Knote: [m[Kdeclared here
   53 | int [01;36m[Kplatform_profile_remove[m[K(struct device *dev);
      |     [01;36m[K^~~~~~~~~~~~~~~~~~~~~~~[m[K
[01m[Klegion-laptop.c:[m[K At top level:
[01m[Klegion-laptop.c:1827:16:[m[K [01;35m[Kwarning: [m[K'[01m[Kecram_memoryio_write[m[K' defined but not used [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wno-unused-function-Wunused-function]8;;[m[K]
 1827 | static ssize_t [01;35m[Kecram_memoryio_write[m[K(const struct ecram_memoryio *ec_memoryio,
      |                [01;35m[K^~~~~~~~~~~~~~~~~~~~[m[K
make[4]: *** [/nix/store/dj7793gb47s632rap69mhigq6lfca5ld-linux-zen-6.14-dev/lib/modules/6.14.0-zen1/source/scripts/Makefile.build:207: legion-laptop.o] Error 1
make[3]: *** [/nix/store/dj7793gb47s632rap69mhigq6lfca5ld-linux-zen-6.14-dev/lib/modules/6.14.0-zen1/source/Makefile:1992: .] Error 2
make[2]: *** [/nix/store/dj7793gb47s632rap69mhigq6lfca5ld-linux-zen-6.14-dev/lib/modules/6.14.0-zen1/source/Makefile:251: __sub-make] Error 2
make[2]: Leaving directory '/build/source/kernel_module'
make[1]: *** [/nix/store/dj7793gb47s632rap69mhigq6lfca5ld-linux-zen-6.14-dev/lib/modules/6.14.0-zen1/source/Makefile:251: __sub-make] Error 2
make[1]: Leaving directory '/nix/store/dj7793gb47s632rap69mhigq6lfca5ld-linux-zen-6.14-dev/lib/modules/6.14.0-zen1/build'
make: *** [Makefile:13: all] Error 2
```

--- a/pkgs/os-specific/linux/lenovo-legion/app.nix
+++ b/pkgs/os-specific/linux/lenovo-legion/app.nix
@@ -8,13 +8,13 @@
 
 python3.pkgs.buildPythonApplication rec {
   pname = "lenovo-legion-app";
-  version = "0.0.19-unstable-2025-02-20";
+  version = "0.0.20-unstable-2025-04-01";
 
   src = fetchFromGitHub {
     owner = "johnfanv2";
     repo = "LenovoLegionLinux";
-    rev = "43f2ad0ea67b69f509a0c5f0254272fe85f3b070";
-    hash = "sha256-k7CHL59tRjfly1xV+TXjNitkhV7q6qtMhuc+QO5ASkI=";
+    rev = "19fef88dbbb077d99052e06f3cd9a3675e7bf3aa";
+    hash = "sha256-0lQ6LyfjZ1/dc6QjB4a1aBcfxY5lIJJEonwuy9a4V4I=";
   };
 
   sourceRoot = "${src.name}/python/legion_linux";

